magento/magento2
# FIX: Can't validate uploaded file for tmp folder mismatching

### Description (*)

when upload file, php $_FILE variable returns destination path of symlinked tmp folder.
but sys_get_temp_dir() function only return target path of symlinked folder.
this mismatch made it can't pass the validation. this commit validate on both target path and destination path if symlink found.
I couldn't find php $_FILE source reference, so I put both path on validation.

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#25835: Invalid parameter given. A valid $fileId[tmp_name] is expected

### Manual testing scenarios (*)

1. set default magento php environment sys_temp_dir ini value to any folder you made symlink to other folder.
2. Go to Stores - Settings - Configuration.
3. Open tab Catalog - Catalog. Fieldset "Product Image Placeholders".
4. Select the placeholder image for field "Base"
5. Click to button "Save"
6. You can try another setting with type "file"
7. check if validation passes or below message will pop.
"Invalid parameter given. A valid $fileId[tmp_name] is expected"

### Questions or comments


### Contribution checklist (*)
